pepper-apis/src/main/java/org/broadinstitute/ddp/security/JWTConverter.java

```diff
@@ -84,25 +94,48 @@ public static String extractEncodedJwtFromHeader(String authHeader) {
         return null;
     }
 
-    private String getPreferredLanguageCodeForUser(Handle handle, String userGuid) {
-        JdbiUser userDao = handle.attach(JdbiUser.class);
-        Long userId = userDao.getUserIdByGuid(userGuid);
-        var profileDao = handle.attach(UserProfileDao.class);
-        UserProfile userProfile = profileDao.findProfileByUserId(userId).orElse(null);
-        String preferredIsoLanguageCode;
-        if (userProfile != null && userProfile.getPreferredLangCode() != null) {
-            preferredIsoLanguageCode = userProfile.getPreferredLangCode();
-            LOG.info("The preferred language code for the user with GUID {} is '{}'",
-                    userGuid, preferredIsoLanguageCode);
-        } else {
-            LOG.info("There is no preferred language code for the user with GUID {}, using a default one ('{}')",
-                    userGuid, DEFAULT_ISO_LANGUAGE_CODE);
-            preferredIsoLanguageCode = DEFAULT_ISO_LANGUAGE_CODE;
+    public JWTConverter() {
+        super();
+        initializeCaching();
+    }
+
+    void resetCaching() {
+        jwtToDDPAuthCache.clear();
+        userIdToJwtCache.clear();
+    }
+
+    private void initializeCaching() {
+        userIdToJwtCache = CacheService.getInstance().getOrCreateCache("userIdToJwtCache",
+                new Duration(MINUTES, 11), this);
+
+
+        jwtToDDPAuthCache = CacheService.getInstance().getOrCreateCache("jwtToDDPAuth",
+                new Duration(MINUTES, 10),
+                (IdToCacheKeyCollectionMapper) (id, handle) -> userIdToJwtCache.get(id), ModelChangeType.USER, this);
+    }
```

Comment:
Here is how we keep the cache from not having invalid entries: we tell the cache service how to map from a user id to to one of our cache keys.
The cache service when it gets a change to for that user, will execute our mapping function and remove that key.